### PR TITLE
Deprecations fixes and minor improvements in gradle scripts

### PR DESCRIPTION
**Pull request by altro3**

### CODE REVIEW

File: buildSrc/src/main/groovy/io.micronaut.build.internal.convention-base.gradle [hunk 1]
```diff
@@ -14,7 +14,7 @@ repositories {
     mavenCentral()
 }
 
-configurations.all {
+configurations.configureEach {
```
Comment (melix) [inline buildSrc/src/main/groovy/io.micronaut.build.internal.convention-base.gradle]: This is actually not required. Configurations cannot be configured lazily. Actually using `configureEach` on the configuration container can lead to hard to debug errors, so I would simply revert these changes.

File: buildSrc/src/main/groovy/io.micronaut.build.internal.convention-quality.gradle [hunk 2]
```diff
@@ -5,7 +5,7 @@ plugins {
 }
 
 tasks.named("htmlSanityCheck") {
-    sourceDir = new File("${rootProject.buildDir}/docs/guide/")
+    sourceDir = file("${rootProject.layout.buildDirectory.asFile.get()}/docs/guide/")
```
Comment (melix) [inline buildSrc/src/main/groovy/io.micronaut.build.internal.convention-quality.gradle]: :x: This is a mix of `file` and `layout`.
```suggestion
        sourceDir = rootProject.layout.buildDirectory.dir("docs/guide").get().asFile
```

```

File: core/build.gradle [hunk 3]
```diff
@@ -31,7 +31,7 @@ spotless {
 }
 
 def versionInfo = tasks.register("micronautVersionInfo", WriteProperties) {
-    outputFile = "${buildDir}/resources/version/micronaut-version.properties"
+    destinationFile = file("${layout.buildDirectory.asFile.get()}/resources/version/micronaut-version.properties")
```
Comment (melix) [inline core/build.gradle]: :x: `destinationFile` is a proper `RegularFileProperty`

```suggestion
    destinationFile = layout.buildDirectory.file("resources/version/micronaut-version.properties")
```

File: test-suite-http-server-tck-netty/build.gradle [hunk 4]
```diff
@@ -48,7 +48,7 @@ graalvmNative {
         enabled = true
     }
     binaries {
-        all {
+        configureEach {
```
Comment (melix) [inline test-suite-http-server-tck-netty/build.gradle]: Same here, `configureEach` doesn't make sense since it's a model element, so it will register tasks in any case. This will never be lazy.

**Review by melix — CHANGES_REQUESTED** [review]
Looks almost good. I would revert the `all` changes which do not make sense and fix the `layout` things.

File: buildSrc/src/main/groovy/io.micronaut.build.internal.convention-base.gradle (hunk 1 above)
Comment (altro3) [reply buildSrc/src/main/groovy/io.micronaut.build.internal.convention-base.gradle]: @melix I didi it, because Idea every time tell me, that is configureEach better than all:

изображение

Comment (melix) [reply buildSrc/src/main/groovy/io.micronaut.build.internal.convention-base.gradle]: Don't always trust your IDE I guess ;)

**Comment by altro3** [comment]
I reverted `all` and fixed layout

File: buildSrc/src/main/groovy/io.micronaut.build.internal.convention-quality.gradle (hunk 2 above)
Comment (altro3) [reply buildSrc/src/main/groovy/io.micronaut.build.internal.convention-quality.gradle]: fixed

File: core/build.gradle (hunk 3 above)
Comment (altro3) [reply core/build.gradle]: fixed

File: test-suite-http-server-tck-netty/build.gradle (hunk 4 above)
Comment (altro3) [reply test-suite-http-server-tck-netty/build.gradle]: reverted

**Review by melix — APPROVED** [review]
